### PR TITLE
Reuse the slack-agent client

### PR DESCRIPTION
**Pull request by kmdkuk**
- Client of slack-agent was being recreated every time (*managerLoop).notifyToSlack was called, so it has been modified to reuse it.
- Fix slack-agent's client to not use Proxy.

Signed-off-by: kouki <kouworld0123@gmail.com>

### CODE REVIEW

File: controllers/runnermanager.go [hunk 1]
```diff
@@ -126,21 +133,30 @@ func newManagerLoop(log logr.Logger, interval time.Duration, k8sClient client.Cl
...
 	m.maxRunnerPods = rp.Spec.MaxRunnerPods
 	m.slackChannel = rp.Spec.SlackAgent.Channel
+	if m.slackAgentServiceName == rp.Spec.SlackAgent.ServiceName {
+		return nil
+	}
+	err := m.slackAgentClient.UpdateServerURL(rp.Spec.SlackAgent.ServiceName)
+	if err != nil {
+		return err
+	}
 	m.slackAgentServiceName = rp.Spec.SlackAgent.ServiceName
```
Comment (masa213f) [inline controllers/runnermanager.go:158]: In this case, it is better to put the update process inside the `if` block.

```suggestion
	if m.slackAgentServiceName != rp.Spec.SlackAgent.ServiceName {
		err := m.slackAgentClient.UpdateServerURL(rp.Spec.SlackAgent.ServiceName)
		if err != nil {
			return err
		}
		m.slackAgentServiceName = rp.Spec.SlackAgent.ServiceName
	}
```

Comment (kmdkuk) [reply controllers/runnermanager.go]: I have reflected it.

**Review by masa213f — APPROVED** [review]
LGTM

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**